### PR TITLE
Mark project command

### PR DESCRIPTION
**Pull request by Johnchiahk**
I tried to do what u said for AddressBook.java, so I did this: 

    public void markProject(Project project) {
        projects.markAsDone(project);
    }
    
    but the status is still not updated to DONE 😓

### CODE REVIEW

File: src/main/java/arb/model/ModelManager.java [hunk 1]
```diff
@@ -107,6 +107,11 @@ public void deleteClient(Client target) {
         addressBook.removeClient(target);
     }
 
+    @Override
+    public void markProject(Project project) {
+        addressBook.markProject(project);
```
Comment (zrei) [inline src/main/java/arb/model/ModelManager.java:112]: I think you need to add one additional line for the project list in the UI to update. Try adding `updateFilteredProjectList(PREDICATE_SHOW_ALL_PROJECTS);` and see if that works?

Comment (Johnchiahk) [reply src/main/java/arb/model/ModelManager.java:112]: It doesn't oops :(

File: src/main/java/arb/logic/commands/project/MarkProjectCommand.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    public CommandResult execute(Model model, ListType currentListBeingShown) throws CommandException {
+        requireNonNull(model);
+        List<Project> lastShownList = model.getFilteredProjectList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PROJECT_DISPLAYED_INDEX);
+        }
+        Project projectToMark = lastShownList.get(targetIndex.getZeroBased());
+        model.markProject(projectToMark);
+
```
Comment (zrei) [inline src/main/java/arb/logic/commands/project/MarkProjectCommand.java:47]: Can you add
```
model.setProject(projectToMark, projectToMark);
model.updatedFilteredProjectList(PREDICATE_SHOW_ALL_PROJECTS);
```
here? And see if it works? If it throws an exception then we'll probably have to make changing Status produce a new Project object.

Comment (Johnchiahk) [reply src/main/java/arb/logic/commands/project/MarkProjectCommand.java:47]: It works holy fishcakes

**Pull request closed without merging**